apps/reaper/test/integration/reaper/reaper_full_test.exs

```diff
@@ -419,6 +431,194 @@ defmodule Reaper.FullTest do
     end
   end
 
+  describe "#{dataset_disable()} then a #{dataset_update()}" do
+    setup %{bypass: bypass} do
+      dataset =
+        TDG.create_dataset(%{
+          technical: %{
+            cadence: "*/5 * * * * * *",
+            sourceUrl: "http://localhost:#{bypass.port}/#{@csv_file_name}",
+            sourceFormat: "csv",
+            sourceType: "stream",
+            schema: [%{name: "id"}, %{name: "name"}, %{name: "pet"}]
+          }
+        })
+
+      Brook.Event.send(@instance, dataset_update(), :reaper, dataset)
+
+      eventually(fn ->
+        assert %{state: :active} = Reaper.Scheduler.find_job(String.to_atom(dataset.id))
+        assert Reaper.Collections.Extractions.is_enabled?(dataset.id) == true
+      end)
+
+      Brook.Event.send(@instance, dataset_disable(), :reaper, dataset)
+
+      eventually(fn ->
+        assert %{state: :inactive} = Reaper.Scheduler.find_job(String.to_atom(dataset.id))
+        assert Reaper.Collections.Extractions.is_enabled?(dataset.id) == false
+      end)
+
+      [dataset: dataset]
+    end
+
+    test "sending an update for the disabled dataset does NOT re-enable it", %{dataset: dataset} do
+      Brook.Event.send(@instance, dataset_update(), :reaper, dataset)
+
+      Process.sleep(5_000)
+
+      eventually(fn ->
+        assert %{state: :inactive} = Reaper.Scheduler.find_job(String.to_atom(dataset.id))
+        assert Reaper.Collections.Extractions.is_enabled?(dataset.id) == false
+      end)
+    end
+  end
+
+  test "dataset:update updates dataset definition in view state", %{bypass: bypass} do
+    dataset =
+      TDG.create_dataset(%{
+        technical: %{
+          cadence: "once",
+          sourceUrl: "http://localhost:#{bypass.port}/#{@csv_file_name}",
+          sourceFormat: "csv",
+          sourceType: "ingest",
+          schema: [%{name: "id"}, %{name: "name"}, %{name: "pet"}]
+        }
+      })
+
+    Brook.Event.send(@instance, dataset_update(), :reaper, dataset)
+
+    eventually(fn ->
+      assert Reaper.Collections.Extractions.get_dataset!(dataset.id) == dataset
+    end)
+  end
+
+  data_test "extracts and ingests update started_timestamp in view state", %{bypass: bypass} do
```

Comment:
Might be nice to name these data_tests so they are reflective of the parameters being used (using `start_event_type ` or `source_type` for instance).

Reply:
Having issues running the integration tests from home but i remember this was an issue with `data_test` not being able to interpolate functions.

Reply:
Yeah, Balser himself noted the data_test caveat here - the output of functions called in the where clause (including macros, somehow) cannot be used in the test name. We could just use the string versions in the where clause and hope it doesn't bite us when/if we update the event type macros. I'm 90% that we won't and that this will be the least of our problems if we do.